cmake/onnxruntime_mlas.cmake

```diff
@@ -348,6 +356,8 @@ else()
           ${MLAS_SRC_DIR}/qgemm_kernel_sdot.cpp
           ${MLAS_SRC_DIR}/sqnbitgemm_kernel_neon.cpp
         )
+        set_source_files_properties(${MLAS_SRC_DIR}/sqnbitgemm_kernel_neon.cpp
+                                    PROPERTIES COMPILE_FLAGS " -march=armv8.2-a+dotprod")
```

Comment:
we still need to support the fp32+int4 on lower arch. Will this break the case?

Reply:
I see you have guard to only run int4 for armv8.2 and above. It is fine to have this assumption.


        // MlasSQNBitGemmDispatchNeon has a dependency on dot product instructions
        this->SQNBitGemmDispatch = &MlasSQNBitGemmDispatchNeon;